numpy/core/src/multiarray/descriptor.c

```diff
@@ -3798,6 +3798,102 @@ _subscript_by_index(PyArray_Descr *self, Py_ssize_t i)
     return _subscript_by_name(self, name);
 }
 
+static npy_bool
+_is_list_of_strings(PyObject *obj)
+{
+    int seqlen, i;
+    if (!PyList_CheckExact(obj)) {
+        return NPY_FALSE;
+    }
+    seqlen = PyList_GET_SIZE(obj);
+    for (i = 0; i < seqlen; i++) {
+        PyObject *item = PyList_GET_ITEM(obj, i);
+        if (!PyBaseString_Check(item)) {
+            return NPY_FALSE;
+        }
+    }
+
+    return NPY_TRUE;
+}
+
+NPY_NO_EXPORT PyArray_Descr *
+arraydescr_field_subset_view(PyArray_Descr *self, PyObject *ind)
+{
+    int seqlen, i;
+    PyObject *fields = NULL;
+    PyObject *names = NULL;
+    PyArray_Descr *view_dtype;
+
+    seqlen = PyList_GET_SIZE(ind);
+
+    fields = PyDict_New();
+    if (fields == NULL) {
+        goto fail;
+    }
+    names = PyTuple_New(seqlen);
+    if (names == NULL) {
+        goto fail;
+    }
+
+    for (i = 0; i < seqlen; i++) {
+        PyObject *name = PyList_GET_ITEM(ind, i);
+        PyObject *tup = PyDict_GetItem(self->fields, name);
+
+        if (tup == NULL) {
+            PyErr_SetObject(PyExc_KeyError, name);
```

Comment:
Was the decision to leave this as `ValueError` until 1.15?

If so, I'll add a `throw_1_13_error` boolean argument to switch between `KeyError` for `dtype.__getitem__` (more correct), and `ValueError` for `array.__getitem__` (backcompat)

Reply:
I'm thinking the change to KeyError and this PR as a whole can wait for 1.15. That way it can go out at the same time as the multifield copy-> view changes (assuming we revert that for 1.14.1).

Reply:
I worry that this will conflict heavily with the fixes to #10411